hummingbot/core/api_throttler/api_request_context_base.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import time
+
+from typing import (
+    Deque,
+)
+
+from hummingbot.core.api_throttler.data_types import (
+    RateLimit,
+    TaskLog,
+    Seconds
+)
+
+
+class APIRequestContextBase:
+
+    def __init__(self,
+                 task_logs: Deque[TaskLog],
+                 pending_tasks: Deque[TaskLog],
+                 rate_limit: RateLimit,
+                 stop_event: asyncio.Event,
+                 period_safety_margin: Seconds = 0.1,
+                 retry_interval: Seconds = 0.1):
+        """
+        Asynchronous context associated with each API request.
+        :param task_logs: Shared task logs
+        :param rate_limit: Rate limit for the associated API request
+        :param rate_limit_type: Rate limit type
+        :param period_safety_margin: estimate for the network latency
+        :param retry_interval: Time between each limit check
+        """
+        self._lock = asyncio.Lock()
+        self._stop_event = stop_event
```

Comment:
I don't understand the use of _stop_event. Is it only used to signal the cancelation of a task?
What happens when it is None because it was not provided as a parameter? (For example the `__init__` function in `FixedRateRequestContext` is not sending the parameter as part of the super() call)

Reply:
I have removed this.